integration-tests/tests/fixtures/package_config_fixtures.py

```diff
@@ -0,0 +1,56 @@
+"""Fixtures that create and remove temporary config files for CLP packages."""
+
+import logging
+from collections.abc import Iterator
+from pathlib import Path
+
+import pytest
+
+from tests.utils.clp_mode_utils import CLP_MODE_CONFIGS
+from tests.utils.config import PackageConfig
+from tests.utils.utils import get_env_var
+
+logger = logging.getLogger(__name__)
+
+
+def _build_package_config_for_mode(mode_name: str) -> PackageConfig:
+    """Construct a PackageConfig for the given `mode_name`."""
+    if mode_name not in CLP_MODE_CONFIGS:
+        err_msg = f"Unknown CLP mode '{mode_name}'."
+        raise KeyError(err_msg)
+
+    clp_package_dir = Path(get_env_var("CLP_PACKAGE_DIR")).expanduser().resolve()
+    test_root_dir = Path(get_env_var("CLP_BUILD_DIR")).expanduser().resolve() / "integration-tests"
+    build_config = CLP_MODE_CONFIGS[mode_name][0]
+    required_components = CLP_MODE_CONFIGS[mode_name][1]
+
+    return PackageConfig(
```

Comment:
I think the name `PackageConfig` has become a bit confusing. My original intent was for this class to represent the configuration of `<clp_root>/build/clp-package`, but it seems you’ve adapted it to handle `clp_config.yml` instead.
I think it'd be nice to have a dataclass called `PackageInstanceClpConfig`, or just move `build_config` and `required_components` to `PackageInstance`